### ISSUE TITLE
Issues with v0.9.0

### ISSUE THREAD

**Issue by harmonf13** [qe-collaborative-services/1115-hub#82]
Issue1:

Some bundles are being received with no Encounter Resource.  An example of the error being returned by the scorecard validator:

"rochester" "2024-03-20" "testcase15" "FAIL" " A scorecard validation error occurred: Validation Test 5: No Encounter resources found"

Issue2:

In the Patient resource there are some null values causing issues.

What you are sending for rochester test case 24:

`                "extension": [
                    null,
                    {
                        "extension": [
                            {
                                "url": "ombCategory",
                                "valueCoding": {
                                    "system": "urn:oid:2.16.840.1.113883.6.238",
                                    "code": "2153-5",
                                    "display": "Mexican American Indian"
                                }
                            }
                        ],
                        "url": "http://hl7.org/fhir/us/core/StructureDefinition/us-core-ethnicity"
                    },
                    null,
                    null
                ],`

If I update to the following it works:

`                "extension": [
                    {
                        "extension": [
                            {
                                "url": "ombCategory",
                                "valueCoding": {
                                    "system": "urn:oid:2.16.840.1.113883.6.238",
                                    "code": "2153-5",
                                    "display": "Mexican American Indian"
                                }
                            }
                        ],
                        "url": "http://hl7.org/fhir/us/core/StructureDefinition/us-core-ethnicity"
                    }
                ],`

Issue 3:

We are getting the error:  A scorecard validation error occurred: Validation Test 16: Patient gender is required

We are seeing with some test cases for rochester, "2024-03-20"	"testcase21" and "2024-03-20 "testcase23" are 2 examples that we provided Patient Gender in the original CSV test files but Gender is missing from the FHIR you sent.

Issue 4:

We are getting the error for several healtheconnections submissions for "2024-03-22"	"testcase6" and "2024-03-22"	"testcase5".  We are seeing an issue with Patient sexual orientations.  It appears there are two issues.  First , it appears the "system" "display" and "code" values are mixed up.  Next the value 38628009 needs to be in "38628009" quotes.

Here is what you are sending:

` "extension": [
                            {
                                "url": "ombCategory",
                                "valueCoding": {
                                    "system": 38628009,
                                    "code": "Homosexual",
                                    "display": "http://snomed.info/sct"
                                }
                            }
                        ],`

@raphaelvp @reeager @shah

**Comment by raphaelvp** [comment]
Issue 1:

Updated the Encounter full url and id in the build 0.10.0. Encounter resource is generated in the bundle only if the EncounterID is present in the source screening CSV file.

The scorecard validator is returning the following error response even if Encounter resource is present in the Bundle.
```
{
  "status": "Invalid resourceType Encounter: at least one encounter required.",
  "bundleID": "92c6f480-eab0-11ee-adb9-016bcb2858b9_healthelink-20240305-testcase3-MRN",
  "test": "Test case routed to scorecard",
  "scoredcardResponse": "QEName: healthelink - TestDate: 20240305 - TestCase: testcase3 - TestResult: PASS"
}
```

Issue 2:

In the Patient resource there are some null values causing issues - The issues related to the null values are still being worked on. 

Issue 3:

As we inspected the source file we could find that the file does not contain the required patient gender information and hence the gender field in FHIR was not populated. Gender Identity is not a required field as per the Omnibus Rules. Please confirm if the Gender Identity is a mandatory field.

Issue 4:

It appears that the source CSV files were having incorrect data which resulted in the wrong mapping for the given fields. Could not investigate further as the source file were not available from the previous build. If possible can you share the source CSV files so that we can verify the same.

@harmonf13 @reeager @shah

**Comment by reeager** [comment]
@raphaelvp We should include an encounter resource no matter what for now even if ID is not present.

**Comment by fredharmonbcbsscgit** [comment]
@raphaelvp 

Issue 4:

Here is the CSV from one of the failing tests related to the issue reported with Patient Sexual Orientation.

MPI_ID|PAT_MRN_ID|FACILITY_ID|CONSENT|FIRST_NAME|MIDDLE_NAME|LAST_NAME|ADMINISTRATIVE_SEX_CODE|ADMINISTRATIVE_SEX _CODE_DESCRIPTION|ADMINISTRATIVE_SEX _CODE_SYSTEM|SEX_AT_BIRTH_CODE|SEX_AT_BIRTH_CODE_DESCRIPTION|SEX_AT_BIRTH_CODE_SYSTEM|PAT_BIRTH_DATE|ADDRESS1|ADDRESS2|CITY|STATE|ZIP|GENDER_IDENTITY_CODE|GENDER_IDENTITY_CODE_DESCRIPTION|GENDER_IDENTITY_CODE_SYSTEM_NAME|SEXUAL_ORIENTATION_CODE|SEXUAL_ORIENTATION_DESCRIPTION|SEXUAL_ORIENTATION_CODE_SYSTEM_NAME|PREFERRED_LANGUAGE_CODE|PREFERRED_LANGUAGE_DESCRIPTION|PREFERRED_LANGUAGE_CODE_SYSTEM_NAME|RACE_CODE|RACE_CODE_DESCRIPTION|RACE_CODE_SYSTEM_NAME|ETHNICITY_CODE|ETHNICITY_CODE_DESCRIPTION|ETHNICITY_CODE_SYSTEM_NAME|MEDICAID_CIN
|healtheconnections-20240322-testcase6-MRN|CNYSCN|No|Jacqueline|Andrea|Carr|X|Undifferentiated|http://terminology.hl7.org/CodeSystem/v3-AdministrativeGender|UNK|unknown|http://terminology.hl7.org/CodeSystem/v3-NullFlavor|1987-04-30|0030 Jason Route|PO Box 34891|Elmira|NY|14901|446151000124109|Male|http://snomed.info/sct|20430005|Heterosexual|http://snomed.info/sct|arw|Arawak|ISO|1056-1|Coast Miwok|urn:oid:2.16.840.1.113883.6.238|2173-3|Uruguayan|urn:oid:2.16.840.1.113883.6.238|TV46617Z

**Comment by n-kes** [comment]
> @raphaelvp @reeager
> 
> Issue 3:
> 
> Here is the CSV that was the original input for the failure related to FHIR Gender. We expect the "ADMINISTRATIVE_SEX*" fields to map to "gender" in the Patient Resource. In Omnibus, for Admin_SEX, the Code and System are reuquired but the description is optional. In the example below, we provided the code and the system but no description. We'd expect the Patient Resource gender to have "X" from the below sample. **I can see what you've got wrong is that you got your wires got cross between "Gender Identity" fields and "Administrative Sex"**. Admin_Sex fields map to "Patient.gender". Gender_Identity_* fields map to Patient.extension (https://hl7.org/fhir/R4/extension-patient-genderidentity.html).
> 
> MPI_ID|PAT_MRN_ID|FACILITY_ID|CONSENT|FIRST_NAME|MIDDLE_NAME|LAST_NAME|ADMINISTRATIVE_SEX_CODE|ADMINISTRATIVE_SEX _CODE_DESCRIPTION|ADMINISTRATIVE_SEX _CODE_SYSTEM|SEX_AT_BIRTH_CODE|SEX_AT_BIRTH_CODE_DESCRIPTION|SEX_AT_BIRTH_CODE_SYSTEM|PAT_BIRTH_DATE|ADDRESS1|ADDRESS2|CITY|STATE|ZIP|GENDER_IDENTITY_CODE|GENDER_IDENTITY_CODE_DESCRIPTION|GENDER_IDENTITY_CODE_SYSTEM_NAME|SEXUAL_ORIENTATION_CODE|SEXUAL_ORIENTATION_DESCRIPTION|SEXUAL_ORIENTATION_CODE_SYSTEM_NAME|PREFERRED_LANGUAGE_CODE|PREFERRED_LANGUAGE_DESCRIPTION|PREFERRED_LANGUAGE_CODE_SYSTEM_NAME|RACE_CODE|RACE_CODE_DESCRIPTION|RACE_CODE_SYSTEM_NAME|ETHNICITY_CODE|ETHNICITY_CODE_DESCRIPTION|ETHNICITY_CODE_SYSTEM_NAME|MEDICAID_CIN |rochester-20240322-testcase21-MRN|CNYSCN|Unk|Nathaniel|Tiffany|Avery|X||[http://terminology.hl7.org/CodeSystem/v3-AdministrativeGender|M||http://terminology.hl7.org/CodeSystem/v3-AdministrativeGender|2010-01-26|85329](http://terminology.hl7.org/CodeSystem/v3-AdministrativeGender%7CM%7C%7Chttp://terminology.hl7.org/CodeSystem/v3-AdministrativeGender%7C2010-01-26%7C85329) Berg Summit|Floor 3 Apt 5|Schenectady|NY|12301||||||[http://snomed.info/sct|bin|Bini](http://snomed.info/sct%7Cbin%7CBini); Edo||1002-5|||2176-6||urn:oid:2.16.840.1.113883.6.238|IA07928O

This issue came up again on a call with @reeager and the Bronx team this afternoon. They can't properly locate those data in the file without having a solid expectation of what the codeset is named. Rachel proposed that we go ahead and fix this now on both the QCS and NYeC sides for both "ADMINISTRATIVE_SEX _CODE_DESCRIPTION" and "ADMINISTRATIVE_SEX _CODE_SYSTEM"

@harmonf13 @raphaelvp

**Comment by raphaelvp** [comment]
Resolved by 0.10.3
